RC 0.3.5

jklenzing opened the pull request:
# Description

RC for v0.3.5

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs (include link [here](https://pysatmissions.readthedocs.io/en/rc_0_3_5/))
- [x] Upload the test distribution to the [testPyPi server](https://test.pypi.org/project/pysatMissions/0.3.5/)
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the release text (put link [here](https://github.com/pysat/pysatMissions/wiki/v0.3.5))
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes
- [x] Update zenodo.json file for new code contributors

If this is a release PR, replace the first item of the above checklist with the
release checklist on the pysat wiki:
https://github.com/pysat/pysat/wiki/Checklist-for-Release

jklenzing committed:
Merge pull request #114 from pysat/main

Update develop with v0.3.4

jklenzing committed:
MAINT: update workflows

jklenzing committed:
DOC: update changelog

jklenzing committed:
MAINT: add sphinx rtd theme cap

jklenzing committed:
MAINT: use pip install

jklenzing committed:
Merge pull request #115 from pysat/maint/workflows

MAINT: update GA workflows

aburrell committed:
DOC: add funding acknowledgements

Added funding acknowledgements to the repository and documentation.

aburrell committed:
DOC: updated changelog

Added a summary of changes to the changelog.

aburrell committed:
DOC: updated init header

Updated the main repository init header.

aburrell committed:
DOC: updated instrument headers

Updated the instrument headers with the NRL pub release.

aburrell committed:
DOC: updated instrument method headers

Updated the Instrument methods headers with the NRL pub release statement.

aburrell committed:
DOC: updated methods headers

Updated the methods headers with the NRL pub release statement and improved the docstrings.

aburrell committed:
DOC: updated test headers

Updated the headers for the unit test files, improved the docstrings, and updated some comments.

aburrell committed:
DOC: updated utils headers

Updated the utils headers to include the NRL pub release.

aburrell committed:
STY: updated docstrings and line length

Updated the docstrings to remove formatting characters and fixed lines that were too long for PEP8.

aburrell committed:
STY: fix formatting

Remove extra lines and capitalize planet's name.

aburrell committed:
DOC: improved instructions

Improved spacing in the documentation, line length in the readme, and added a
link to the full package documentation in the documentation section of the
readme.

aburrell committed:
MAINT: removed extra figure

Removed the extra "poweredbypysat" figure from the main directory.

aburrell committed:
STY: removed trailing whitespace

Removed trailing whitespace from the docstring.

aburrell committed:
MAINT: added .readthedocs.yml

Added a yaml for ReadTheDocs.

aburrell committed:
Merge pull request #118 from pysat/nrl_pub_release

NRL pub release

jklenzing committed:
MAINT: update workflow versions

jklenzing committed:
DOC: update changelog

jklenzing committed:
DOC: update pyproject.toml

jklenzing committed:
BUG: pip install

jklenzing committed:
Merge pull request #119 from pysat/maint/nep29_apr24

jklenzing committed:
DOC: tentative release date

rstoneback reviewed (commented):
Looks good! Sorry for the delay. Saw this review request in my messages today.

One request, can my institution in the zenodo document be updated to Cosmic Studio ?

jklenzing committed:
Update .zenodo.json

jklenzing commented:
> Looks good! Sorry for the delay. Saw this review request in my messages today.
> 
> One request, can my institution in the zenodo document be updated to Cosmic Studio ?

Thanks for catching that. Fixed it.

jklenzing commented on CHANGELOG.md:
```diff
@@ -2,6 +2,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](https://semver.org/).
 
+## [0.3.5] - 2024-05-10
```
```suggestion
## [0.3.5] - 2024-07-12
```

jklenzing committed:
Update CHANGELOG.md

jklenzing commented:
@aburrell, I was updating the date to finish this off and found a new apexpy related bug:
```
E           NameError: name 'fa' is not defined
```

aburrell commented:
Looks like that could be related to the release of numpy 2.0.0 😿.  It'll probably (?) be fixed if you limit the numpy version to 1.X.

On a more positive note, I think if you specify the compilers, you can get apexpy installed on the MacOS tests.

aburrell committed:
TST: apexpy installation

Try to fix apexpy installation.

aburrell committed:
BUG: remove tabs

Remove tabs from yaml.

aburrell committed:
BUG: fix apexpy errors

Fix errors in apexpy installation attempt.

aburrell committed:
BUG: uninstall numpy

Added an uninstall line for numpy to fix downgrade.  Also fixed the gcc calls.

aburrell committed:
BUG: replaced tab

Replaced tab with spaces.

aburrell committed:
BUG: find gcc

Added lines to find gcc and a potential fix for uninstalling numpy.

aburrell committed:
BUG: fix numpy install

Fix the limited numpy install to use a good max version number and changed the gcc debug statements.

aburrell committed:
BUG: remove numpy fix

Remove the numpy fix attempt

aburrell committed:
BUG: remove tab

Removed tab from yaml.

aburrell committed:
BUG: toggle apexpy to NEP29

Only test apexpy with oldest numpy.

aburrell committed:
BUG: numpy is being upgraded

The 'only-if-needed' is upgrading numpy to latest.

aburrell committed:
BUG: change numpy install order

Try changing the numpy install order.

aburrell committed:
BUG: downgrade scipy

Downgrade scipy for NEP29.

aburrell commented:
Ok, I fixed the MacOS installation, but couldn't get GA to downgrade numpy for the apexpy install.  So, I toggled current testing to NEP29 only. This lead to the uncovering of a bug, where you were never testing NEP29.  So I also fixed NEP29.

The apexpy numpy 2.0.0 issues are documented, and it can run with numpy 2.0.0 if you install with a lower version of numpy.  So we'll work on that for a new release.

jklenzing commented on CHANGELOG.md:
```diff
@@ -2,6 +2,13 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](https://semver.org/).
 
+## [0.3.5] - 2024-07-12
```
```suggestion
## [0.3.5] - 2024-07-16
```

jklenzing committed:
Update CHANGELOG.md
